files/image_config/monit/memory_checker

```diff
@@ -59,6 +62,87 @@ def get_command_result(command):
 
     return command_stdout.strip()
 
+def get_container_id(container_name):
+    """Gets full container ID of the specified container
+    Args:
+        container_name: A string indicates the name of specified container.
+    Returns:
+        container_id: A string indicates the full ID of specified container.
+    """
+    container_id = ""
+
+    get_container_info_cmd = "docker ps --no-trunc"
+    command_stdout = get_command_result(get_container_info_cmd)
+
+    for line in command_stdout.splitlines():
+        if container_name in line:
+            container_id = line.split()[0].strip()
+            break
+
+    if not container_id:
+        syslog.syslog(syslog.LOG_ERR, "[memory_checker] Failed to get contianer ID of '{}'! Exiting ..."
+                      .format(container_name))
+        sys.exit(4)
+
+    return container_id
+
+def get_memory_usage(container_id):
+    """Reads the container's memory usage from the control group subsystem's file
+    '/sys/fs/cgroup/memory/docker/<container_id>/memory.usage_in_bytes'.
+    Args:
+        container_id: A string indicates the full ID of a container.
+    Returns:
+        memory_usage_in_bytes: A string indicates memory usage (Bytes) of a container.
+    """
+    memory_usage_in_bytes = ""
+
+    docker_memory_usage_file_path = CGROUP_DOCKER_MEMORY_DIR + container_id + "/memory.usage_in_bytes"
+    if not os.path.exists(docker_memory_usage_file_path):
+        syslog.syslog(syslog.LOG_ERR,
+                      "[memory_checker] cgroup memory usage file '{}' of container '{}' does not exist on device! Exiting ..."
+                      .format(docker_memory_usage_file_path, container_id))
+        sys.exit(5)
+
+    get_memory_usage_cmd = "sudo cat {}".format(docker_memory_usage_file_path)
+    memory_usage_in_bytes = get_command_result(get_memory_usage_cmd)
+    if not memory_usage_in_bytes:
+        syslog.syslog(syslog.LOG_ERR, "[memory_checker] Failed to get the memory usage of container '{}'! Exiting ..."
+                      .format(container_id))
+        sys.exit(6)
+
+    return memory_usage_in_bytes
+
+def get_cache_usage(container_id):
```

Comment:
I think maybe we can refer to this as getting inactive cache usage since we want to calculate container memory usage as total memory usage - inactive cache usage

Reply:
updated.